### PR TITLE
docs: add Seattle annual temperature area trellis example

### PR DESCRIPTION
**Pull request by domoritz**
by @axiomabsolute

### CODE REVIEW

File: examples/compiled/trellis_area_seattle.vg.json [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+          "as": "row_hours_date_sort_index"
```
Comment (domoritz) [inline examples/compiled/trellis_area_seattle.vg.json]: Why is this here twice?

**Comment by jakevdp** [comment]
One comment: rather than specifying sort directly, what do you think about using a calculate transform with a sort field?
```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "description": "Average temperatures in Seattle, broken down by hour. Derived from [Seattle Annual Temperate](https://vega.github.io/vega/examples/annual-temperature/) example from the Vega example gallery.",
  "title": "Seattle Annual Temperatures",
  "data": {"url": "data/seattle-temps.csv"},
  "transform": [
    {"calculate": "(hours(datum.date) + 18) % 24", "as": "order"}
  ],
  "config": {"axis": {"grid": false, "domain": false}},
  "spacing": {"row": 1},
  "facet": {
    "row": {
      "field": "date",
      "timeUnit": "hours",
      "type": "nominal",
      "sort": {"field": "order"},
      "header": {
        "labelAngle": 0,
        "labelPadding": 2,
        "titlePadding": -4,
        "labelAlign": "left",
        "labelExpr": "hours(datum.value) == 0 ? 'Midnight' : hours(datum.value) == 12 ? 'Noon' : timeFormat(datum.value, '%I:%M %p')"
      }
    }
  },
  "spec": {
    "width": 800,
    "height": 25,
    "view": {"stroke": null},
    "mark": "area",
    "encoding": {
      "x": {
        "field": "date",
        "type": "temporal",
        "title": "Month",
        "axis": {"labelExpr": "timeFormat(datum.value, '%b')"}
      },
      "y": {
        "field": "temp",
        "type": "quantitative",
        "axis": {"title": null, "labels": false, "ticks": false}
      }
    }
  }
}
```

**Comment by domoritz** [comment]
Great idea. I want to also look into why the previous example generates two calculates, which I separated out into https://github.com/vega/vega-lite/issues/6069.

**Comment by domoritz** [comment]
I removed the warning about length encoding marks (area, bar) with zero.
